### ISSUE TITLE
Skymatch step documentation seems incorrect

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#6161]
_Issue [JP-2154](https://jira.stsci.edu/browse/JP-2154) was created on JIRA by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra):_

The documentation of the skymatch step of the pipeline ([here|#limitations-and-discussions]]) states: 

"Since sky subtraction is performed on flat-fielded but not distortion corrected images, it is important to keep in mind that flat-fielding is performed to obtain uniform surface brightness and not flux. This distinction is important for images that have not been distortion corrected. As a consequence, it is advisable that point-like sources be masked through the user-supplied mask files.”

I am not sure this part is correct. The pixel-area map is already applied to the stage-2 cal.fits images, thus the value of each pixel already takes into account for the effects of the distortion. Even if this is not true, the problem/limitation described above would be also true for a pixel that is not part of a point source but, for example, of the background.

Could these lines be something left from the HST pipeline documentation ([here](https://stsci-skypac.readthedocs.io/en/latest/skymatch.html))? For HST, flc/flt images are flat-fielded but not distortion corrected and their pixel values are not corrected for pixel area. Thus, the HST pipeline needs to correct for the pixel-area differences when creating the drc/drz images.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533931#comment-533931):_

There's been a lot of discussion over the years of these various processes that account, in one way or another, for varying pixel size due to distortion. First, the flat-field step does in fact correct for variations in pixel area, due to the fact that the flat-field reference file itself is in unrectified/distorted space and hence pixels with a larger area have higher signal in the flat (assuming the illumination for the flat is itself perfectly flat). So when the flatfield image gets divided into a science image the surface brightness levels in each pixel will now be correct, or at least normalized, to provide constant surface brightness across the image. However, consider the case of 2 unresolved point sources having equal intrinsic brightness, with 1 falling on a pixel having larger area than the pixel in which the 2nd source is contained. Assuming no differences in pixel sensitivity, those 2 pixels would have equal signal levels before flat-fielding. But after flat-fielding, the signal in the pixel with the larger area is now reduced relative the other one. So in the flat-fielded image the 2 "equal" point sources no longer have equal signal levels. So if this situation occurs in the overlap region of 2 images being used by the skymatch step, i.e. the same point source falling on different detector pixels in the 2 images, they now show up with different signal levels when doing the comparison of the overlap region and will bias the matching algorithm in skymatch.

Regarding the pixel area map: it's true that the pixel area map reference file is **attached** to the cal product during the `photom` step of `calwebb_image2`, but it's not actually applied to the data in any way. It's there for information only, which can be used during analysis if performing photometry on the unrectified/distorted cal products, because - as described above - photometry done on equal point sources in cal images will result in different values and hence need additional correction.

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) please chime in and correct any errors I've made in the above description!

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533938#comment-533938):_

While I agree with the statements above, in practice I think that the differences are quite small.  In addition, for the skymatch step, a sigma clipping is done to determine the levels in each overlap image for comparison, no?  This will remove point sources and, hence, minimize their impact on this step.  In addition, it is not clear if a discussion of user supplied files is appropriate for this pipeline step.  I do feel that this step should work well without user supplied files and this is not what the documentation implies.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533939#comment-533939):_

Yes, I agree that in general the flat field does not correct for the different pixel sizes due to the geometric distortion but only for the response of the pixels. However, my understanding is that for JWST (or at least for MIRI) the pixel-area map is applied to the flat field. If so, pixel values in stage-2 images should be corrected for area differences, even though they are still in the distorted frame. This is not an issue because then the calwebb_image3 resample step should just compensate for the fact that the surface-brightness value of a pixel in the stage-3 image is a "combination" of the surface-brightness values of various pixels in the stage-2 image (I'm assuming there is only 1 image).

Anyway, let's wait for [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) reply.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533965#comment-533965):_

Ah, yes, you may be right about the pixel area being included in the MIRI flat-fields (or is it in the photom ref files?), unlike other instruments. [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) may also be able to weigh in on this.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533977#comment-533977):_

Pixel area is included in **all** the flat fields for all the instruments.  That has been the plan all along.  If this is not the case, then the reference files need updating.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534012#comment-534012):_

I remember this came up recently; <https://jira.stsci.edu/browse/JWSTMIRI-196> redelivered MIRI imager pixel flats so that they DID include the pixel area correction as Karl said.  As far as I'm aware MRS is the only mode where this isn't true yet (flats are all unity, and all the relevant corrections are baked into PHOTOM for reasons irrelevant to the present discussion).

I'm not familiar with the algorithm in use by SKYMATCH, but at least for the equivalent task MRS_IMATCH I know there's a sigma clipping in place to deal with stars in the overlap region.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534018#comment-534018):_

Just to recap my original comment. The pixel-area map is already applied to the stage-2 cal.fits files when performing the flat-fielding. Thus, the statement in the readthedocs should be modified. The stage-2 images are not distortion corrected, but the surface-brightness values in each pixel are. Thus, the pipeline can compare the stage-2 images in the skymatch step.

In general, if the pixel-area correction is not applied, all pixels are not OK for the task, regardless if they are part of a point source, an extended source or just background pixels.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534024#comment-534024):_

OK, I guess we're down to the point of just arguing over semantics. The variation in pixel area is accounted for when dividing by the flat-field, but it happens automatically by simply using the raw relative counts in the flat-field images as is, which means pixels with larger area have larger values in the flat-field. There's no explicit application or inclusion of an actual independent pixel area map either in the flat-field image itself or when the flat-field image is applied to the science data. It just happens automatically due to the way the flats are constructed. This is the same for both HST and JWST flats - they both use flats that are in "raw" detector space, where signals vary in proportion to pixel area. So I would argue that saying "the pixel area map is applied" is not correct and misleading. No pixel area map is applied. All that's applied is the flat-field image.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534056#comment-534056):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) comment above, first paragraph in the first comment at the top, is exactly correct. The issue is in how sky matching is performed. In skypac.skymatch matching is done on the mean pixel brightness in the overlapping regions. Since regions must overlap on the sky, total area of the regions in two images is the same. MRS_IMATCH uses a different algorithm and matching is done on intensity of each pixel (pixel-by-pixel) on drizzled (undistorted) and aligned (stackable) images so a pixel with the same image coordinates in all images has identical sky area and corresponds to the same point on the sky. Both algorithms (skymatch and mrs_imatch) expect as input flat-fielded images.

 

Also see <https://hst-docs.stsci.edu/wfc3dhb/chapter-9-wfc3-data-analysis/9-1-wfc3-data-analysis#id-9.1WFC3DataAnalysis-9.1.119.1.11PixelAreaMaps>

 

I *think* the effect of "wrong" flat-fielded intensities will be negligible (due to error averaging and due to point sources being fewer than background pixels). However, I do not believe this makes that paragraph in the docs incorrect.

 

In any case, let me know if you want docs corrected or feel free to make a PR.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534800#comment-534800):_

Hi [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) and [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara), thanks for the explanation. You are right, the mistake was on my side. I understood that the pixel-area map was applied at a certain point by the pipeline, but this is not correct. As you said, the surface brightness is preserved, but not the flux. Thus, as discussed in the HST link above, one should multiply by the pixel-area map when working with point sources (although there might be another layer of complexity with MIRI since the flat fields have been normalized to the average pixel area). For this part of the ticket I have opened, there is no action needed.

Karl had some thoughts about the masking (see his first comment), but I let him propose how to move forward.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534926#comment-534926):_

I wonder if this discussion has gotten a bit off-topic.  For sky-match, sigma-clipping is used in calculating the offset between two images.  This is there to avoid biases due to cosmic rays and point sources (i.e., sampling issues).  Given sigma-clipping is included, then the discussion of point sources and flat fields while quite interesting, is not needed for the sky-match step.  In this case, maybe the statement on masking of point sources can be removed/changed to say it might improve the matching?

Also, the statement in the sky-match documentation that flat fielding gives correct surface brightness values, but not fluxes should be removed.  The flat fielding is producing the **correct** surface brightness for all sources regardless of size (a pixel's measurement does not care nor "know" the source size).  When extracting measurements for a source, then the issues due to sampling and distortion come into play.  As the text currently reads, it implies that the surface brightnesses are correct, but the fluxes are not after flat fielding.  This is just not true.  The images are correctly calibrated.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=564945#comment-564945):_

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) This discussion is stale and it wandered a few times, but if I can summarize, the net requested change for this ticket is to remove:

```java
Because sky subtraction is performed on flat-fielded but not distortion corrected images, it is important to keep in mind that flat-fielding is performed to obtain uniform surface brightness and not flux.
```

And possibly replace with

```java
Because sky subtraction is performed on flat-fielded but not distortion corrected images, it is important to keep in mind that flat-fielding is performed to obtain correct surface brightness across all pixel sizes. Extracted source measurements will be affected by sampling and distortion, however, and so distortion corrections will be necessary before such extraction occurs.
```

?

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=564986#comment-564986):_

Sounds good.  One suggested change would be "...across all pixel sizes." to "...for all pixels.".  The "sizes" bit is not really important and may be  a bit distracting.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565010#comment-565010):_

I am sorry, but I disagree with most of the changes. I can agree only with the following change:

```java
uniform surface brightness
```

to

```java
correct surface brightness
```

And even this change is weird. What is correct here? Of course "uniform surface brightness" without context is not better. I think the idea was that a true uniform source on the sky (a perfectly uniform background, as an example) would look uniform in the image (as opposite to having a non-uniform surface brightness caused by different pixel sized (in srad) due to distortions). I think the fact that "uniform surface brightness" is expected for truly uniform real sources was missing here.

The second sentence is confusing and I do not know what is the point of it in the documentation of skymatch. For example, the original paragraph's purpose was to explain fine points in how masking should be done. This second sentence seems to be teaching people how to extract "source measurements" and distortion-correction before extractions on undistorted images. Many people extract source measurements from distorted images. I just do not want to introduce in this paragraph a discussion about that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565038#comment-565038):_

If minimal changes are desired I would advocate changing "..to obtain uniform surface brightness and not flux." to "...to obtain correct surface brightnesses."  (this includes Mihai's suggested change).

The whole discussion of fluxes is fine to leave out.  I just want to say that the surface brightness and fluxes are **correct** after flat fielding.  They have to be as the pixels have no knowledge of the size of the source.  Only when one wishes to extract a flux of a specific source do the distortions need to be taken into account.  I agree that discussing how to extract sources from distorted images is best left to other documentation.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565045#comment-565045):_

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) The minimal change results in a somewhat nonsensical reading - I've added what I hope to be acceptable wording for all involved, and we can put this paragraph (and ticket) behind us. Please take a look when you can.

**Comment by karllark** [comment]
Change proposed in linked PR looks good to me.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565051#comment-565051):_

1. What is "correct" needs to be explained in the context of the issue (sky matching procedure). For example, flat field correction really involves two corrections: one for the sensitivity variations between pixels and one related to their sky coverage (pixel scale variation exclusively due to distortion). If we do not apply this last correction (due to distortion) then the flat-file will still be "correct" (i.e., nothing wrong with it) but the processing like "source measurement" will be different. I think should best be explained in the data handbook-like documentation that describes specifics of individual data products (RAW, FLT, DRZ, etc.) What I was saying in my previous comment is that the relevant for sky matching algorithm effect of flat-fielding is constant pixel intensity for a constant (uniform in space) real source in the image regardless of its extent or position in the image and a possible side-effect for point sources.

2. I do not see how second sentence adds any clarity as related to sky matching. I do not think we should be discussing "extracted source measurements" as it has nothing to do with sky matching.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565053#comment-565053):_

[Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) It's not clear to me that you are looking at the most recent commit to the PR. I will reproduce it here:


```java
  Because sky subtraction is performed on *flat-fielded* but
   *not distortion corrected* images, it is important to keep in mind
   that flat-fielding is performed to obtain correct surface brightnesses.
   As the surface brightness of a pixel containing a point-like source will
   change inversely with a change to the pixel area, it is advisable that
   point-like sources be masked through the user-supplied mask files. Values
   different from zero in user-supplied masks indicate good data pixels.
   Alternatively, one can use the ``upper`` parameter to limit the use of
   pixels containing bright objects in the sky computations.
```

The additional sentence I have added attempts to explain what correct means, in the context of point-like sources, and why masking is then relevant. There is no longer any mention of extracted sources. Please confirm this commit is also unsatisfactory.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565054#comment-565054):_

Sorry, didn't realize there was a PR.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565055#comment-565055):_

It's perfect. Thanks!

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565060#comment-565060):_

Fixed by #6546

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533931#comment-533931):_

There's been a lot of discussion over the years of these various processes that account, in one way or another, for varying pixel size due to distortion. First, the flat-field step does in fact correct for variations in pixel area, due to the fact that the flat-field reference file itself is in unrectified/distorted space and hence pixels with a larger area have higher signal in the flat (assuming the illumination for the flat is itself perfectly flat). So when the flatfield image gets divided into a science image the surface brightness levels in each pixel will now be correct, or at least normalized, to provide constant surface brightness across the image. However, consider the case of 2 unresolved point sources having equal intrinsic brightness, with 1 falling on a pixel having larger area than the pixel in which the 2nd source is contained. Assuming no differences in pixel sensitivity, those 2 pixels would have equal signal levels before flat-fielding. But after flat-fielding, the signal in the pixel with the larger area is now reduced relative the other one. So in the flat-fielded image the 2 "equal" point sources no longer have equal signal levels. So if this situation occurs in the overlap region of 2 images being used by the skymatch step, i.e. the same point source falling on different detector pixels in the 2 images, they now show up with different signal levels when doing the comparison of the overlap region and will bias the matching algorithm in skymatch.

Regarding the pixel area map: it's true that the pixel area map reference file is **attached** to the cal product during the `photom` step of `calwebb_image2`, but it's not actually applied to the data in any way. It's there for information only, which can be used during analysis if performing photometry on the unrectified/distorted cal products, because - as described above - photometry done on equal point sources in cal images will result in different values and hence need additional correction.

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) please chime in and correct any errors I've made in the above description!

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533938#comment-533938):_

While I agree with the statements above, in practice I think that the differences are quite small.  In addition, for the skymatch step, a sigma clipping is done to determine the levels in each overlap image for comparison, no?  This will remove point sources and, hence, minimize their impact on this step.  In addition, it is not clear if a discussion of user supplied files is appropriate for this pipeline step.  I do feel that this step should work well without user supplied files and this is not what the documentation implies.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533939#comment-533939):_

Yes, I agree that in general the flat field does not correct for the different pixel sizes due to the geometric distortion but only for the response of the pixels. However, my understanding is that for JWST (or at least for MIRI) the pixel-area map is applied to the flat field. If so, pixel values in stage-2 images should be corrected for area differences, even though they are still in the distorted frame. This is not an issue because then the calwebb_image3 resample step should just compensate for the fact that the surface-brightness value of a pixel in the stage-3 image is a "combination" of the surface-brightness values of various pixels in the stage-2 image (I'm assuming there is only 1 image).

Anyway, let's wait for [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) reply.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533965#comment-533965):_

Ah, yes, you may be right about the pixel area being included in the MIRI flat-fields (or is it in the photom ref files?), unlike other instruments. [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) may also be able to weigh in on this.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=533977#comment-533977):_

Pixel area is included in **all** the flat fields for all the instruments.  That has been the plan all along.  If this is not the case, then the reference files need updating.

**Comment by stscijgbot-jp** [comment]
_Comment by [David Law](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dlaw) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534012#comment-534012):_

I remember this came up recently; <https://jira.stsci.edu/browse/JWSTMIRI-196> redelivered MIRI imager pixel flats so that they DID include the pixel area correction as Karl said.  As far as I'm aware MRS is the only mode where this isn't true yet (flats are all unity, and all the relevant corrections are baked into PHOTOM for reasons irrelevant to the present discussion).

I'm not familiar with the algorithm in use by SKYMATCH, but at least for the equivalent task MRS_IMATCH I know there's a sigma clipping in place to deal with stars in the overlap region.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534018#comment-534018):_

Just to recap my original comment. The pixel-area map is already applied to the stage-2 cal.fits files when performing the flat-fielding. Thus, the statement in the readthedocs should be modified. The stage-2 images are not distortion corrected, but the surface-brightness values in each pixel are. Thus, the pipeline can compare the stage-2 images in the skymatch step.

In general, if the pixel-area correction is not applied, all pixels are not OK for the task, regardless if they are part of a point source, an extended source or just background pixels.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534024#comment-534024):_

OK, I guess we're down to the point of just arguing over semantics. The variation in pixel area is accounted for when dividing by the flat-field, but it happens automatically by simply using the raw relative counts in the flat-field images as is, which means pixels with larger area have larger values in the flat-field. There's no explicit application or inclusion of an actual independent pixel area map either in the flat-field image itself or when the flat-field image is applied to the science data. It just happens automatically due to the way the flats are constructed. This is the same for both HST and JWST flats - they both use flats that are in "raw" detector space, where signals vary in proportion to pixel area. So I would argue that saying "the pixel area map is applied" is not correct and misleading. No pixel area map is applied. All that's applied is the flat-field image.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534056#comment-534056):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) comment above, first paragraph in the first comment at the top, is exactly correct. The issue is in how sky matching is performed. In skypac.skymatch matching is done on the mean pixel brightness in the overlapping regions. Since regions must overlap on the sky, total area of the regions in two images is the same. MRS_IMATCH uses a different algorithm and matching is done on intensity of each pixel (pixel-by-pixel) on drizzled (undistorted) and aligned (stackable) images so a pixel with the same image coordinates in all images has identical sky area and corresponds to the same point on the sky. Both algorithms (skymatch and mrs_imatch) expect as input flat-fielded images.

 

Also see <https://hst-docs.stsci.edu/wfc3dhb/chapter-9-wfc3-data-analysis/9-1-wfc3-data-analysis#id-9.1WFC3DataAnalysis-9.1.119.1.11PixelAreaMaps>

 

I *think* the effect of "wrong" flat-fielded intensities will be negligible (due to error averaging and due to point sources being fewer than background pixels). However, I do not believe this makes that paragraph in the docs incorrect.

 

In any case, let me know if you want docs corrected or feel free to make a PR.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mattia Libralato](https://jira.stsci.edu/secure/ViewProfile.jspa?name=libra) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534800#comment-534800):_

Hi [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) and [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara), thanks for the explanation. You are right, the mistake was on my side. I understood that the pixel-area map was applied at a certain point by the pipeline, but this is not correct. As you said, the surface brightness is preserved, but not the flux. Thus, as discussed in the HST link above, one should multiply by the pixel-area map when working with point sources (although there might be another layer of complexity with MIRI since the flat fields have been normalized to the average pixel area). For this part of the ticket I have opened, there is no action needed.

Karl had some thoughts about the masking (see his first comment), but I let him propose how to move forward.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=534926#comment-534926):_

I wonder if this discussion has gotten a bit off-topic.  For sky-match, sigma-clipping is used in calculating the offset between two images.  This is there to avoid biases due to cosmic rays and point sources (i.e., sampling issues).  Given sigma-clipping is included, then the discussion of point sources and flat fields while quite interesting, is not needed for the sky-match step.  In this case, maybe the statement on masking of point sources can be removed/changed to say it might improve the matching?

Also, the statement in the sky-match documentation that flat fielding gives correct surface brightness values, but not fluxes should be removed.  The flat fielding is producing the **correct** surface brightness for all sources regardless of size (a pixel's measurement does not care nor "know" the source size).  When extracting measurements for a source, then the issues due to sampling and distortion come into play.  As the text currently reads, it implies that the surface brightnesses are correct, but the fluxes are not after flat fielding.  This is just not true.  The images are correctly calibrated.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=564945#comment-564945):_

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) This discussion is now quite stale, but if I can summarize, the net requested change for this ticket is to remove:

```java
Since sky subtraction is performed on flat-fielded but not distortion corrected images, it is important to keep in mind that flat-fielding is performed to obtain uniform surface brightness and not flux.
```

And possibly replace with

```java
Because sky subtraction is performed on flat-fielded but not distortion corrected images, it is important to keep in mind that flat-fielding is performed to obtain correct surface brightness across all pixel sizes. Extracted source measurements will be affected by sampling and distortion, however, and so distortion corrections will be necessary before such extraction occurs.
```

?

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=564986#comment-564986):_

Sounds good.  One suggested change would be "...across all pixel sizes." to "...for all pixels.".  The "sizes" bit is not really important and may be  a bit distracting.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565010#comment-565010):_

I am sorry, but I disagree with most of the changes. I can agree only with the following change:

```java
uniform surface brightness
```

to

```java
correct surface brightness
```

And even this change is weird. What is correct here? Of course "uniform surface brightness" without context is not better. I think the idea was that a true uniform source on the sky (a perfectly uniform background, as an example) would look uniform in the image (as opposite to having a non-uniform surface brightness caused by different pixel sized (in srad) due to distortions). I think the fact that "uniform surface brightness" is expected for truly uniform real sources was missing here.

The second sentence is confusing and I do not know what is the point of it in the documentation of skymatch. For example, the original paragraph's purpose was to explain fine points in how masking should be done. This second sentence seems to be teaching people how to extract "source measurements" and distortion-correction before extractions on undistorted images. Many people extract source measurements from distorted images. I just do not want to introduce in this paragraph a discussion about that.

**Comment by stscijgbot-jp** [comment]
_Comment by [Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565038#comment-565038):_

If minimal changes are desired I would advocate changing "..to obtain uniform surface brightness and not flux." to "...to obtain correct surface brightnesses."  (this includes Mihai's suggested change).

The whole discussion of fluxes is fine to leave out.  I just want to say that the surface brightness and fluxes are **correct** after flat fielding.  They have to be as the pixels have no knowledge of the size of the source.  Only when one wishes to extract a flux of a specific source do the distortions need to be taken into account.  I agree that discussing how to extract sources from distorted images is best left to other documentation.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565045#comment-565045):_

[Karl Gordon](https://jira.stsci.edu/secure/ViewProfile.jspa?name=kgordon) [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) The minimal change results in a somewhat nonsensical reading - I've added what I hope to be acceptable wording for all involved, and we can put this paragraph (and ticket) behind us. Please take a look when you can.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565051#comment-565051):_

1. What is "correct" needs to be explained in the context of the issue (sky matching procedure). For example, flat field correction really involves two corrections: one for the sensitivity variations between pixels and one related to their sky coverage (pixel scale variation exclusively due to distortion). If we do not apply this last correction (due to distortion) then the flat-file will still be "correct" (i.e., nothing wrong with it) but the processing like "source measurement" will be different. I think should best be explained in the data handbook-like documentation that describes specifics of individual data products (RAW, FLT, DRZ, etc.) What I was saying in my previous comment is that the relevant for sky matching algorithm effect of flat-fielding is constant pixel intensity for a constant (uniform in space) real source in the image regardless of its extent or position in the image and a possible side-effect for point sources.

2. I do not see how second sentence adds any clarity as related to sky matching. I do not think we should be discussing "extracted source measurements" as it has nothing to do with sky matching.

**Comment by stscijgbot-jp** [comment]
_Comment by [Tyler Pauly](https://jira.stsci.edu/secure/ViewProfile.jspa?name=tpauly) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565053#comment-565053):_

[Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) It's not clear to me that you are looking at the most recent commit to the PR. I will reproduce it here:


```java
  Because sky subtraction is performed on *flat-fielded* but
   *not distortion corrected* images, it is important to keep in mind
   that flat-fielding is performed to obtain correct surface brightnesses.
   As the surface brightness of a pixel containing a point-like source will
   change inversely with a change to the pixel area, it is advisable that
   point-like sources be masked through the user-supplied mask files. Values
   different from zero in user-supplied masks indicate good data pixels.
   Alternatively, one can use the ``upper`` parameter to limit the use of
   pixels containing bright objects in the sky computations.
```

The additional sentence I have added attempts to explain what correct means, in the context of point-like sources, and why masking is then relevant. There is no longer any mention of extracted sources. Please confirm this commit is also unsatisfactory.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565054#comment-565054):_

Sorry, didn't realize there was a PR.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565055#comment-565055):_

It's perfect. Thanks!

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2154?focusedCommentId=565060#comment-565060):_

Fixed by #6546
